pkg/util/id.go

```diff
@@ -16,31 +16,81 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
 	"io/ioutil"
+	"os"
+	"path"
 	"strings"
+
+	"github.com/google/uuid"
+)
+
+const (
+	maxNodeIDLength = 12
 )
 
-const maxMachineIDLength = 12
+// To ensure a constant node ID between runs, the id will be persisted in the data directory.
+func LoadPersistedNodeID(dataDir string) (string, error) {
+	id, err := loadNodeID(dataDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to load node id. %+v", err)
+	}
 
-func GetMachineID() (string, error) {
-	buf, err := ioutil.ReadFile("/etc/machine-id")
+	err = saveNodeID(dataDir, id)
```

Comment:
if the id is already on disk this will load it and immediately save it, right? we dont need to save it if its already there.